faas/faas.go

```diff
@@ -15,29 +15,117 @@
 package faas
 
 import (
-	"os"
+	"context"
+	"fmt"
 
-	"github.com/valyala/fasthttp"
+	"github.com/nitrictech/go-sdk/constants"
+	pb "github.com/nitrictech/go-sdk/interfaces/nitric/v1"
+	"google.golang.org/grpc"
 )
 
 // NitricFunction - a function built using Nitric, to be executed
-type NitricFunction func(*NitricRequest) *NitricResponse
+type NitricFunction func(*NitricTrigger) (*NitricResponse, error)
+
+func faasLoop(stream pb.Faas_TriggerStreamClient, f NitricFunction, errorCh chan error) {
+	for {
+		// Block recieving a message
+		srvrMsg, err := stream.Recv()
+		clientMsg := &pb.ClientMessage{
+			Id: srvrMsg.GetId(),
+		}
+
+		if err != nil {
+			// TODO: Make sure we use the correct kind of error types here
+			errorCh <- err
+			break
+		}
+
+		// We have a trigger
+		if srvrMsg.GetTriggerRequest() != nil {
+			req, err := FromGrpcTriggerRequest(srvrMsg.GetTriggerRequest())
+
+			if err != nil {
+				fmt.Println("There was an error reading the TriggerRequest", err)
+				// Return a bad request here...
+
+				continue
+			}
+			// Let the membrane know the function is ready for initializatio
```

Comment:
Should this comment still be here? If yes, it has a typo at the end.

Reply:
Fixed.